sagemaker-python-sdk/tensorflow_script_mode_mnist/tensorflow_distributed_mnist.ipynb

```diff
@@ -0,0 +1,224 @@
+    "\n",
+    "Here is the entire script:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "!cat 'mnist.py'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a training job using the sagemaker.TensorFlow estimator\n",
+    "\n",
+    "The ``sagemaker.TensorFlow`` estimator handles locating the script mode container uploading your script to a S3 location and creating a SageMaker training job."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "from sagemaker.tensorflow import TensorFlow\n",
+    "\n",
+    "\n",
+    "mnist_estimator = TensorFlow(entry_point='mnist.py',\n",
+    "                             role='SageMakerRole',\n",
+    "                             train_instance_count=2,\n",
+    "                             train_instance_type='ml.p2.xlarge',\n",
+    "                             framework_version='1.12',\n",
+    "                             py_version = 'py3',\n",
+    "                             distributions = {'parameter_server': {'enabled': True}},\n",
+    "                             base_job_name='test-tf')\n",
+    "\n",
+    "mnist_estimator.fit(training_data_uri)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The **```fit```** method will create a training job in two **ml.p2.xlarge** instances. The logs above will show the instances doing training, evaluation, and incrementing the number of **training steps**. \n",
+    "\n",
+    "In the end of the training, the training job will generate a saved model for TF serving."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "# Deploy the trained model to prepare for predictions\n",
+    "\n",
+    "The deploy() method creates an SageMaker model which is then used to create an endpoint which serves prediction requests in realtime. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor = mnist_estimator.deploy(initial_instance_count=1, instance_type='ml.p2.8xlarge')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Invoking the endpoint\n",
+    "\n",
+    "Let's donwload the training data and use that as input for inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "!aws s3 cp s3://sagemaker-sample-data-us-west-2/tensorflow/mnist/train_data.npy train_data.npy\n",
+    "!aws s3 cp s3://sagemaker-sample-data-us-west-2/tensorflow/mnist/train_labels.npy train_labels.npy\n",
```

Comment:
the region part should be dynamic. also you might need to put `--region` in the command to make sure it works everywhere (it might be easier to use https for GovCloud)